### ISSUE TITLE
Adding a space before the first element crashes the editor

### ISSUE THREAD

**Issue by unparalloser** [spyder-ide/spyder#14912]
### What steps will reproduce the problem?

If anything that can be autocompleted (print, import, def, ...) is entered as the very first element of a document, adding a space before it and then deleting the space will make Spyder encounter an internal problem.

### Traceback
```python-traceback
  File "/nix/store/w2xg4mpza8mny71c15hg44b0d82xxcbg-python3.8-spyder-4.2.0/lib/python3.8/site-packages/spyder/plugins/completion/snippets/actor.py", line 92, in handle_msg
    if language_snippets[current_word]:
  File "/nix/store/w2xg4mpza8mny71c15hg44b0d82xxcbg-python3.8-spyder-4.2.0/lib/python3.8/site-packages/spyder/plugins/completion/snippets/trie.py", line 81, in __getitem__
    if len(sequence) == 0:
TypeError: object of type 'NoneType' has no len()
```

## Versions

* Spyder version: 4.2.0 
* Python version: 3.8.7
* Qt version: 5.15.2
* PyQt5 version: 5.15.2
* Operating System: Linux 5.4.99

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  3.0.4 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=4.0                   :  7.19.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  21.5.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.8.0 (NOK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.7.2 (OK)
pylint >=1.0                    :  2.6.2 (OK)
pyls >=0.36.1;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.1.1             :  0.3.0 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.1 (OK)
qtconsole >=4.7.7               :  5.0.1 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
setuptools >=39.0.0             :  50.3.1.post0 (OK)
sphinx >=0.6.6                  :  3.3.1 (OK)
spyder_kernels >=1.10.0;<1.11.0 :  1.10.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  2.0.0 (OK)
xdg >=0.26                      :  0.27 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  None (NOK)
matplotlib >=2.0.0              :  3.3.3 (OK)
numpy >=1.7                     :  1.19.4 (OK)
pandas >=1.1.1                  :  None (NOK)
scipy >=0.17.0                  :  1.6.0 (OK)
sympy >=0.7.3                   :  None (NOK)
```

**Comment by ccordoba12** [comment]
Hey @unparalloser, thanks for reporting. This error is fixed in our latest version (4.2.3). Please update and let us know if it fixes the problem for you.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
